packages/java-edition/src/mcfunction/mcdocAttributes.ts

```diff
@@ -0,0 +1,85 @@
+import * as core from '@spyglassmc/core'
+import { localize } from '@spyglassmc/locales'
+import * as mcdoc from '@spyglassmc/mcdoc'
+import * as mcf from '@spyglassmc/mcfunction'
+import { getItemSlotsArgumentValues } from './common/index.js'
+import {
+	EntitySelectorAtVariable,
+	EntitySelectorNode,
+	ScoreHolderNode,
+} from './node/argument.js'
+import * as parser from './parser/index.js'
+
+export function registerMcdocAttributes(
+	meta: core.MetaRegistry,
+	rootTreeNode: mcf.RootTreeNode,
+) {
+	mcdoc.runtime.registerAttribute(meta, 'command', () => undefined, {
+		// TODO: validate slash
+		// TODO: fix completer inside commands
+		stringParser: () => {
+			return mcf.command(rootTreeNode, parser.argument)
+		},
+	})
+	mcdoc.runtime.registerAttribute(meta, 'text_component', () => undefined, {
```

Comment:
How are inline SNBT handled in e.g. JSON strings? I could see some kind of inline reference here coupled with the normal JSON parser instead of hardcoding text_component specifically here

Reply:
We have a `nbt` attribute that takes a mcdoc type, so yeah we could replace `text_component` with a `json` attribute doing something similar.